### DISCUSSION TITLE
openssl_pkey_export(): Cannot get key from parameter 1

### DISCUSSION THREAD

**Discussion by goigarg** [spatie/crypto#31] [Q&A]
i am getting the error while generating key
 [$privateKey, $publicKey] = (new KeyPair())->generate();

**Comment by giagara** [comment]
same here, running on lambda with bref. Php 8.2.

Did you get any workaround?

**Comment by whykhamist** [comment]
### For anyone having this issue and on windows, here's what I did:

1. Search `env` in the start menu
2. click on `Edit the system environment variables`
3. On the `System Properties` window under `Advanced` tab, click on `Environment Variables`
4. Under `System Variables` click on `New`
5. Enter the `Variable name` OPENSSL_CONF
6. Enter the `Variable value`: 
   - Wamp - C:\wamp\bin\apache\Apache2.2.17\conf\openssl.cnf
   - Xampp - C:\xampp\apache\conf\openssl.cnf
7. Click "OK" and close all the windows and RESTART your computer.

Source: https://stackoverflow.com/a/18869750
